launcher/minecraft/mod/tasks/ModFolderLoadTask.cpp

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2024
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ModFolderLoadTask.h"
+
+#include "Application.h"
+#include "FileSystem.h"
+#include "minecraft/mod/MetadataHandler.h"
+
+#include <QThread>
+
+static bool isModFile(const QFileInfo& info)
+{
+    if (!info.isFile()) {
+        return false;
+    }
+
+    QString name = info.fileName().toLower();
+    if (name.endsWith(".disabled")) {
+        name.chop(9);
+    }
+
+    return name.endsWith(".jar") || name.endsWith(".zip") || name.endsWith(".litemod") || name.endsWith(".nilmod");
+}
+
+ModFolderLoadTask::ModFolderLoadTask(const QDir& root_dir,
+                                     bool is_indexed,
+                                     bool clean_orphan,
+                                     std::function<Resource*(const QFileInfo&)> create_function)
+    : Task(false)
+    , m_rootDir(root_dir)
+    , m_isIndexed(is_indexed)
+    , m_cleanOrphan(clean_orphan)
+    , m_createFunc(create_function)
+    , m_result(new Result())
+    , m_threadToSpawnInto(thread())
+{}
+
+void ModFolderLoadTask::executeTask()
+{
+    if (thread() != m_threadToSpawnInto) {
+        connect(this, &Task::finished, this->thread(), &QThread::quit);
+    }
+
+    if (m_isIndexed) {
+        getFromMetadata(m_rootDir);
+    }
+
+    getFromFiles(m_rootDir, 0);
+
+    if (m_cleanOrphan) {
+        QMutableMapIterator iter(m_result->resources);
+        while (iter.hasNext()) {
+            auto resource = iter.next().value();
+            if (resource->status() == ResourceStatus::NOT_INSTALLED) {
+                QDir index_dir(QDir(resource->fileinfo().absolutePath()).filePath(".index"));
+                resource->destroy(index_dir, false, false);
+                iter.remove();
+            }
+        }
+    }
+
+    for (auto mod : m_result->resources) {
+        mod->moveToThread(m_threadToSpawnInto);
+    }
+
+    if (m_aborted) {
+        emit finished();
```

Comment:
This line is bad: 
you should call emitAborted() function or at least have an additional call to ` emit aborted();` 
as we expect a task to finish in one of the 3 states: success,fail or abort 

Disclaimer I do understand that the `ResourceFolderLoadTask` has a similar issue: feel free to fix it there too but that is not necessary into this PR.

Disclaimer no2 I realize task  will never be aborted but still we should keep the same logic.